code/backend/batch/AddURLEmbeddings.py

```diff
@@ -1,20 +1,21 @@
 import logging
 import traceback
 import azure.functions as func
-import sys
 
 from utilities.helpers.DocumentProcessorHelper import DocumentProcessor
 from utilities.helpers.ConfigHelper import ConfigHelper
 
-sys.path.append("..")
 
 bp_add_url_embeddings = func.Blueprint()
-
 logger = logging.getLogger(__name__)
 
 
 @bp_add_url_embeddings.route(route="AddURLEmbeddings")
 def add_url_embeddings(req: func.HttpRequest) -> func.HttpResponse:
+    return do_add_url_embeddings(req)
```

Comment:
Why is this required? Can't you just import `add_url_embeddings` in the tests?

Reply:
No, it doesn't work, I suspect because of the decorator

Reply:
Ah I think you can make it work by first "building" the function. I just followed this guide and it worked for one of the tests https://medium.com/mesh-ai-technology-and-engineering/writing-and-testing-azure-functions-in-the-v2-python-programming-model-c391bd779ff6

Reply:
By "doesn't work" I mean that the function call simply won't execute from the test, you can try to change this and set the breakpoint inside add_url_embeddings when running the test - it won't stop there

Reply:
Is that even following the guide above?

I only tried it with one test, but the below worked for me:
```python
@patch("backend.batch.AddURLEmbeddings.ConfigHelper")
@patch("backend.batch.AddURLEmbeddings.DocumentProcessor")
def test_add_url_embeddings_when_url_set_in_body(_, __):
    fake_request = func.HttpRequest(
        method="POST",
        url="",
        body=b'{"url": "https://example.com"}',
        headers={"Content-Type": "application/json"},
    )

    func_call = add_url_embeddings.build().get_user_function() # Note this line <----------
    response = func_call(fake_request)

    assert response.status_code == 200
```

Reply:
Sorry I hadn't seen your reply when I typed in the comment, will try this

Reply:
The method is now triggered but weirdly all paths for `patch` call have broken now

Reply:
^ looks like it was my fault for breaking the patch, this has been applied now so I'm going to resolve this thread